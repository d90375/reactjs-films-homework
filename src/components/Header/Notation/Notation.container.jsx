import React, { useState } from "react";
import PropTypes from "prop-types";
import Notation from "./Notation";

const NotationContainer = React.memo(({ movieId, overview }) => {
  const [isHiddenViewWindow, setIsHiddenViewWindow] = useState(true);

  const handleToggleView = () => {
    setIsHiddenViewWindow(!isHiddenViewWindow);
  };
  return (
    <>
      <Notation
        movieId={movieId}
        overview={overview}
        isHiddenViewWindow={isHiddenViewWindow}
        onToggleView={handleToggleView}
      />
    </>
  );
});

export default NotationContainer;

NotationContainer.displayName = "NotationContainer";

NotationContainer.propTypes = {
  movieId: PropTypes.number,
  overview: PropTypes.string
};

NotationContainer.defaultProps = {
  movieId: 0,
  overview: ""
};
