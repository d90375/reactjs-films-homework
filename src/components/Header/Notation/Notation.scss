.flexFixedWrap {
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
}

.notation {
  display: flex;
  flex-direction: column;
  align-items: flex-end;
  margin-right: 20px;
}

.notation__wrap {
  width: calc(270px - 0.5rem * 2);
  height: calc(135px - 0.5rem);
  overflow: hidden;
  text-overflow: clip;
  background-color: rgba(0, 0, 0, 0.64);
  padding: 0.5rem 1rem 0 1rem;
  margin-bottom: 2rem;
  text-align: justify;
}

.notation__btn {
  align-self: flex-start;
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  margin-right: 1.5rem;
}

.notation__text {
  color: #fff;
  font-family: "Helvetica Neue - thin", serif;
  font-size: 0.9rem;
  line-height: 1.5rem;
  letter-spacing: 1px;
}

.btn {
  width: 8rem;
  height: 2rem;
  margin-right: 1.5rem;
  text-align: center;
  line-height: 20px;
  outline: none;
  font-size: 1rem;
  font-family: "Helvetica Neue - thin", serif;
  font-weight: 400;
  letter-spacing: 1.3px;
  color: #fff;
  cursor: pointer;
}

.btn__watch {
  background-color: #0aaee4;
  border: 1px solid #0aaee4;
  border-radius: 5px;

  &:hover {
    background: lighten(#0aaee4, 10%);
  }
}

.btn__view {
  background-color: transparent;
  border: 1px solid #fff;
  border-radius: 5px;

  &:hover {
    text-shadow: 0 0 10px rgba(255, 255, 255, 1);
  }
}
