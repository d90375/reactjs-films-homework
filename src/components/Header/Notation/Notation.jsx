import PropTypes from "prop-types";
import React from "react";
import "./Notation.scss";
import ButtonWatchNow from "../../VideoFrame/ButtonWatchNow/ButtonWatchNow";

const Notation = ({ movieId, overview, isHiddenViewWindow, onHideView }) => {
  return (
    <div className="flexFixedWrap">
      <div className="notation">
        {isHiddenViewWindow && (
          <div className="notation__wrap">
            <span className="notation__text">{overview}</span>
          </div>
        )}
        <div className="notation__btn">
          <ButtonWatchNow movieId={movieId} />
          <button onClick={onHideView} type="button" className="btn btn__view">
            {isHiddenViewWindow ? "Hide Info" : "View Info"}
          </button>
        </div>
      </div>
    </div>
  );
};

export default Notation;

Notation.propTypes = {
  isHiddenViewWindow: PropTypes.bool,
  movieId: PropTypes.number,
  onHideView: PropTypes.func,
  overview: PropTypes.string
};

Notation.defaultProps = {
  isHiddenViewWindow: false,
  movieId: 0,
  onHideView: () => {},
  overview: ""
};
