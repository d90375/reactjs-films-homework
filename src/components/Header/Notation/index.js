import Notation from "./Notation.container";

export default Notation;
