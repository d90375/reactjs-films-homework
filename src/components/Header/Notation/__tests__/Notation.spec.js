import React from "react";
import Notation from "../Notation";

const setUp = (props) => {
  return shallow(<Notation {...props} />);
};

describe("Notation component", () => {
  let component;

  beforeEach(() => {
    component = setUp();
  });

  const mockProps = {
    movieId: 999,
    isHiddenViewWindow: true,
    overview: "overview"
  };

  it("should render Notation component without props", () => {
    expect(component).toMatchSnapshot();
  });

  it("should render Notation component with props", () => {
    component = setUp({ ...mockProps });
    expect(component).toMatchSnapshot();
  });

  describe("should handler", () => {
    it("should call onHideView method", () => {
      const mockCallBack = jest.fn();
      component = setUp({ onHideView: mockCallBack });
      expect(mockCallBack.mock.calls.length).toBe(0);
      component.find(".btn__view").simulate("click");
      expect(mockCallBack.mock.calls.length).toBe(1);
    });
  });
});
