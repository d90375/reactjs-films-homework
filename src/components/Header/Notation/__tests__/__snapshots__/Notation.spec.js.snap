// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notation component should render Notation component with props 1`] = `
<div
  className="flexFixedWrap"
>
  <div
    className="notation"
  >
    <div
      className="notation__wrap"
    >
      <span
        className="notation__text"
      >
        overview
      </span>
    </div>
    <div
      className="notation__btn"
    >
      <Memo(ButtonWatchNow)
        movieId={999}
      />
      <button
        className="btn btn__view"
        onClick={[Function]}
        type="button"
      >
        Hide Info
      </button>
    </div>
  </div>
</div>
`;

exports[`Notation component should render Notation component without props 1`] = `
<div
  className="flexFixedWrap"
>
  <div
    className="notation"
  >
    <div
      className="notation__btn"
    >
      <Memo(ButtonWatchNow)
        movieId={0}
      />
      <button
        className="btn btn__view"
        onClick={[Function]}
        type="button"
      >
        View Info
      </button>
    </div>
  </div>
</div>
`;
